linera-storage-service/src/server.rs

```diff
@@ -126,21 +137,68 @@ impl SharedStoreServer {
 
     pub async fn clear_journal(&self, base_key: &[u8]) -> Result<(), Status> {
         match self {
-            SharedStoreServer::Memory(store) => store
+            ServiceStoreServer::Memory(store) => store
                 .clear_journal(base_key)
                 .await
                 .map_err(|_e| Status::not_found("clear_journal")),
             #[cfg(feature = "rocksdb")]
-            SharedStoreServer::RocksDb(store) => store
+            ServiceStoreServer::RocksDb(store) => store
                 .clear_journal(base_key)
                 .await
                 .map_err(|_e| Status::not_found("clear_journal")),
         }
     }
+
+    pub async fn list_all(&self) -> Result<Vec<Vec<u8>>, Status> {
+        self.find_keys_by_prefix(&[KeyTag::Namespace as u8]).await
+    }
+
+    pub async fn delete_all(&self) -> Result<(), Status> {
+        let mut batch = Batch::new();
+        batch.delete_key_prefix(vec![KeyTag::Key as u8]);
+        batch.delete_key_prefix(vec![KeyTag::Namespace as u8]);
+        let base_key = vec![];
+        self.write_batch(batch, &base_key).await?;
+        self.clear_journal(&base_key).await
+    }
+
+    pub async fn exist_namespace(&self, namespace: &[u8]) -> Result<bool, Status> {
+        let mut full_key = vec![KeyTag::Namespace as u8];
+        full_key.extend(namespace);
+        self.contains_key(&full_key).await
+    }
+
+    pub async fn create_namespace(&self, namespace: &[u8]) -> Result<(), Status> {
+        let mut full_key = vec![KeyTag::Namespace as u8];
+        full_key.extend(namespace);
+        let mut batch = Batch::new();
+        batch.put_key_value_bytes(full_key, vec![]);
+        let base_key = vec![];
+        self.write_batch(batch, &base_key).await?;
+        self.clear_journal(&base_key).await
+    }
+
+    pub async fn delete_namespace(&self, namespace: &[u8]) -> Result<(), Status> {
+        let mut batch = Batch::new();
+        let mut full_key = vec![KeyTag::Namespace as u8];
+        full_key.extend(namespace);
+        batch.delete_key(full_key);
+        let mut key_prefix = vec![KeyTag::Key as u8];
+        key_prefix.extend(namespace);
+        batch.delete_key_prefix(key_prefix);
+        let base_key = vec![];
+        self.write_batch(batch, &base_key).await?;
+        self.clear_journal(&base_key).await
```

Comment:
Doesn't this interfere with ongoing writes in other namespaces?

Reply:
Yes, so I remove it.
To be precise, the problem occurs because the `base_key` are colliding.
It does not matter since for `Memory` and `RocksDb` the `clear_journal` is trivial.

Reply:
What happens if the journal contains both entries for this and another namespace? Isn't it necessary to delete exactly those for the deleted namespace?

Reply:
I agree that some problems could occur. But for `Memory` and `RocksDb` we do not use any journal.